ci/environment-3.6.yml

```diff
@@ -12,6 +12,7 @@ dependencies:
   - psutil
   - matplotlib
   - flake8
+  - mypy=0.782
```

Comment:
Is there a reason to pin this so tightly?

Reply:
I would prefer no pin

Reply:
Yes. Mypy versions break the checks often

Reply:
Yeah, so shouldn't that be part of the test suite?

Reply:
it is very much like a linter than a test suit. as you see here, it breaks between versions. 
https://github.com/pypa/pip/pull/5792

so it is better to pin them. It is a not library requirement that we specifiy in requirements.txt . so it is safe to pin.

Reply:
Yeah, I just don't want the maintenance burden of having to update the pins

Reply:
if we use poetry, manging dependencies will not be a pain. Maybe I will open another PR if you want for that.

Reply:
Is there anything we can do here to turn this into some sort of valid range?

Reply:
it is a development dependency. do we really need to make it flexible ? we can make range like < 0.8 or something. 
but it doesn't follow semantic versioning.